tests/providers/amazon/aws/sensors/test_s3_key.py

```diff
@@ -121,3 +121,30 @@ def test_poke_wildcard(self, mock_hook):
 
         mock_check_for_wildcard_key.return_value = True
         self.assertTrue(op.poke(None))
+
+
+class TestS3KeySizeSensor(unittest.TestCase):
+    @mock.patch('airflow.providers.amazon.aws.sensors.s3_key.S3Hook')
+    def test_poke(self, mock_hook):
+        op = S3KeySizeSensor(task_id='s3_key_sensor', bucket_key='s3://test_bucket/file')
+
+        mock_check_for_key = mock_hook.return_value.check_for_key
+        mock_check_for_key.return_value = False
+        self.assertFalse(op.poke(None))
+
+        mock_check_for_key.assert_called_once_with(op.bucket_key, op.bucket_name)
+        mock_hook.return_value.check_for_key.return_value = True
+        self.assertFalse(op.poke(None))
+
+        mock_paginator = mock.Mock()
+        mock_paginator.paginate.return_value = []
+        mock_conn = mock.Mock()
+        mock_conn.return_value.get_paginator.return_value = mock_paginator
+        mock_hook.return_value.get_conn = mock_conn
+        mock_paginator.paginate.return_value = [{"Contents": [{"Size": 0}]}]
+        self.assertFalse(op.poke(None))
+
+        mock_paginator.paginate.return_value = [{"Contents": [{"Size": 10}]}]
+        self.assertTrue(op.poke(None))
+        mock_check_for_key.return_value = False
+        self.assertFalse(op.poke(None))
```

Comment:
Please split these tests into separate test functions. This makes it a lot clearer what you test.

Reply:
@feluelle Please review